dspace-xmlui/src/main/java/org/dspace/app/xmlui/aspect/discovery/Navigation.java

```diff
@@ -118,15 +136,80 @@ public void addOptions(Options options) throws SAXException, WingException,
             }
         }
         browseGlobal.addItem().addXref(contextPath + "/community-list", T_head_all_of_dspace );
-    */
+        */
 
         /* regulate the ordering */
         options.addList("discovery");
         options.addList("browse");
         options.addList("account");
-        options.addList("context");
         options.addList("administrative");
-
+                
+        // get uri to see if using discovery and if under a specific handle
+        String uri = request.getSitemapURI();
+        
+        // check value in dspace.cfg
+        String search_export_config = ConfigurationManager.getProperty("xmlui.search.metadata_export"); 
+        
+        // get query
+        String query = decodeFromURL(request.getParameter("query"));
+                
+        // get scope, if not under handle returns null
+        String scope= request.getParameter("scope");
+        
+        // used to serialize all query filters together
+    	String filters = "";
+    	
+    	// get all query filters
+    	String[] fqs = DiscoveryUIUtils.getFilterQueries(ObjectModelHelper.getRequest(objectModel), context);
+        
+    	if (fqs != null)
+        {
+        	for(int i = 0; i < fqs.length; i++) {
+            	if(i < fqs.length - 1)
+            		filters += fqs[i] + ",";
+            	else
+            		filters += fqs[i];
+            }
+        }
+    	
+        if(uri.contains("discover")) {
+        	// check scope
+        	if(scope == null || "".equals(scope))
+        		scope = "/";
+        	// check query
+        	if(query == null || "".equals(query))
+        		query = "*";
+        	// check if under a handle, already in discovery
+        	if(uri.contains("handle")) {
+        		scope = uri.replace("handle/", "").replace("/discover", "");
+            }
+        	// replace forward slash to pass through sitemap
+        	try {
+            	scope = scope.replace("/", "~");
+            }
+            catch(NullPointerException e) { }
+        	if(search_export_config != null) {
+        		// some logging
+        		
+    			log.info("uri: " + uri);
+    			log.info("query: " + query);
+    			log.info("scope: " + scope);
+    			log.info("filters: " + filters);
+        		        		
+        		if(search_export_config.equals("admin")) {
+        			if(authorizeService.isAdmin(context)) {
+        				List results = options.addList("context");    		
+                    	results.setHead(T_context_head);
+                    	results.addItem().addXref(contextPath + "/discover/csv/" + query + "/" + scope + "/" + filters, T_export_metadata);
```

Comment:
Would it not be better to send the query, scope & filters along as request params instead of path params ? What if the search query contains an illegal path character ?